src/request/reducers/RequestDetailRequestReducer.ts

```diff
@@ -92,20 +75,37 @@ function correlateMiddlewareMessages(startMessages: IMessageEnvelope<IMiddleware
 function toMap<T, TResult>(values: T[], keySelector: (value: T) => string, valueSelector: (value: T) => TResult): { [key: string]: TResult } {
 
     return _.reduce(
-        values, 
+        values,
         (result: { [key: string]: TResult }, value: T) => {
             result[keySelector(value)] = valueSelector(value);
 
             return result;
-        }, 
+        },
         <{ [key: string]: TResult }>{});
 }
```

Comment:
To an extent, it was just an exercise in seeing whether I could generally replicate the `ToDictionary()` extension method in .NET, as I was unable to find an equivalent in `lodash` (where you can take an array and create a map using independent selectors for key and value, all in a single step).  It mostly works, but the TypeScript compiler seems finicky about the selectors (e.g. it can't seem to infer the generic type of the method if you pass a predefined selector rather than an inline lambda, even if they're both typed the same way).